core/src/sumeragi/main_loop.rs

```diff
@@ -1184,66 +1094,82 @@ enum BlockSyncError {
 }
 
 fn handle_block_sync(
-    block: VersionedCommittedBlock,
+    block: VersionedSignedBlock,
+    topology: &Topology,
     wsv: &WorldStateView,
     finalized_wsv: &WorldStateView,
-) -> Result<BlockSyncOk, BlockSyncError> {
-    let block_height = block.as_v1().header.height;
+) -> Result<BlockSyncOk, (VersionedSignedBlock, BlockSyncError)> {
+    let block_height = block.payload().header.height;
     let wsv_height = wsv.height();
     if wsv_height + 1 == block_height {
         // Normal branch for adding new block on top of current
         let mut new_wsv = wsv.clone();
-        block
-            .revalidate(&mut new_wsv)
-            .map(|_| BlockSyncOk::CommitBlock(block, new_wsv))
-            .map_err(BlockSyncError::BlockNotValid)
+        ValidBlock::validate_without_topology(block, &mut new_wsv)
```

Comment:
I will create an issue for this. If a peer is unregistered and then registered again and there has been a view change in between the new peer refuses to accept blocks received via block sync

relevant tests are in `connected_peers.rs` and `unstable_network.rs`

Reply:
created #3890